lib/Core/SQLiteBuildDB.cpp

```diff
@@ -41,11 +41,18 @@ class SQLiteBuildDB : public BuildDB {
   /// * 4: Pre-history
   static const int currentSchemaVersion = 7;
 
+  /// Internal key ID type used for linking rule dependencies. Related to, but
+  /// distinct from the build engine's KeyID type.
+  typedef uint64_t DBKeyID;
```

Comment:
This isn't actually going to give us any type safety, is it (because of C++)? I think for us to get type safety we would have to make a wrapper type.

Reply:
True enough.